modules/storage-rocksdb/src/test/java/org/apache/ignite/internal/storage/rocksdb/benchmarks/CommitManyWritesBenchmark.java

```diff
@@ -0,0 +1,213 @@
+
+        var tableDescriptor = new StorageTableDescriptor(TABLE_ID, NUM_PARTITIONS, STORAGE_PROFILE_NAME);
+
+        tableStorage = storageEngine.createMvTable(tableDescriptor, indexId -> null);
+
+        CompletableFuture<?>[] createFutures = IntStream.range(0, NUM_PARTITIONS)
+                .mapToObj(tableStorage::createMvPartition)
+                .toArray(CompletableFuture[]::new);
+
+        CompletableFuture.allOf(createFutures).join();
+    }
+
+    /** Tear down method. */
+    @TearDown
+    public void tearDown() {
+        tableStorage.destroy();
+
+        storageEngine.stop();
+    }
+
+    /** Generated data for each thread. */
+    @State(Scope.Thread)
+    public static class GeneratedData {
+        int partitionId;
+
+        Map<RowId, BinaryRow> rows;
+
+        /** Setup method. */
+        @Setup
+        public void setUp() {
+            ThreadLocalRandom random = ThreadLocalRandom.current();
+
+            partitionId = random.nextInt(NUM_PARTITIONS);
+
+            rows = new HashMap<>(capacity(NUM_ROWS));
+
+            for (int i = 0; i < NUM_ROWS; i++) {
+                var rowId = new RowId(partitionId);
+
+                BinaryRow row = randomRow(random);
+
+                rows.put(rowId, row);
+            }
+        }
+
+        private static BinaryRow randomRow(ThreadLocalRandom random) {
+            ByteBuffer buffer = ByteBuffer.allocate(ROW_LENGTH);
+
+            random.nextBytes(buffer.array());
+
+            return new BinaryRowImpl(0, buffer);
+        }
+    }
+
+    private static RocksDbStorageEngineConfiguration engineConfiguration() {
+        RocksDbStorageEngineConfiguration config = mock(RocksDbStorageEngineConfiguration.class);
+
+        ConfigurationValue<Integer> flushDelayMillis = mock(ConfigurationValue.class);
+
+        when(flushDelayMillis.value()).thenReturn(100);
+
+        when(config.flushDelayMillis()).thenReturn(flushDelayMillis);
+
+        return config;
+    }
+
+    private static StorageConfiguration storageConfiguration() {
+        StorageConfiguration config = mock(StorageConfiguration.class);
+
+        NamedConfigurationTree profilesTree = mock(NamedConfigurationTree.class);
+        NamedListView profilesView = mock(NamedListView.class);
+        RocksDbProfileView rocksDbProfileView = mock(RocksDbProfileView.class);
+
+        when(rocksDbProfileView.name()).thenReturn(STORAGE_PROFILE_NAME);
+        when(rocksDbProfileView.size()).thenReturn(16777216L);
+        when(rocksDbProfileView.writeBufferSize()).thenReturn(16777216L);
+
+        when(config.profiles()).thenReturn(profilesTree);
+        when(profilesTree.value()).thenReturn(profilesView);
+        when(profilesView.iterator()).thenReturn(List.of(rocksDbProfileView).iterator());
+
+        return config;
+    }
+
+    /** Benchmark. */
+    @Benchmark
+    public void commitManyWrites(GeneratedData data) {
+        MvPartitionStorage partitionStorage = tableStorage.getMvPartition(data.partitionId);
```

Comment:
We could recreate storage after every iteration

Reply:
What will it change? I think this way it tests a more realistic situation (when RocksDb has data in it)